timdex_dataset_api/utils.py

```diff
@@ -84,3 +98,90 @@ def _split_s3_uri(s3_uri: str) -> tuple[str, str]:
         bucket = parsed.netloc
         key = parsed.path.lstrip("/")  # strip leading slash from /key
         return bucket, key
+
+
+def sa_reflect_duckdb_conn(
+    conn: DuckDBPyConnection, schema: str | None = None
+) -> MetaData:
+    """Use reflection to return SQLAlchemy metadata about a DuckDB connection.
+
+    Args:
+        - conn: DuckDB connection
+        - schema: if provided, schema to reflect from; default of None results in the
+        DuckDB 'main' schema
+    """
+    start_time = time.perf_counter()
+    db_metadata = MetaData()
+
+    engine = create_engine(
+        "duckdb://",
+        creator=lambda: ConnectionWrapper(conn),
+    )
```

Comment:
This is a bit of a hack, but kind of unsure another way to do it.  We need a SQLAlchemy `engine` object that we can use for reflection/introspection... but we already have a DuckDB connection, we don't want this to do that for us.

This was inspired by [this comment](https://github.com/Mause/duckdb_engine/issues/1143#issuecomment-2440048837) from the `duckdb_engine` SQLAlchemy library creator.

What's important to remember is that we don't use this engine for anything _except_ introspection, so not terribly critical.